commands/project/list/list.go

```diff
@@ -0,0 +1,95 @@
+package list
+
+import (
+	"fmt"
+
+	"github.com/profclems/glab/pkg/iostreams"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/profclems/glab/commands/cmdutils"
+	"github.com/profclems/glab/internal/glrepo"
+	"github.com/profclems/glab/pkg/tableprinter"
+	"github.com/spf13/cobra"
+	"github.com/xanzy/go-gitlab"
+)
+
+type Options struct {
+	OrderBy string
+	Sort    string
+	PerPage int
+	Page    int
+
+	BaseRepo   func() (glrepo.Interface, error)
+	HTTPClient func() (*gitlab.Client, error)
+	IO         *iostreams.IOStreams
+}
+
+func NewCmdList(f *cmdutils.Factory) *cobra.Command {
+	opts := &Options{
+		IO: f.IO,
+	}
+	var repoListCmd = &cobra.Command{
+		Use:   "list",
+		Short: `Get list of your repositories.`,
+		Example: heredoc.Doc(`
+	$ glab repo list
+	`),
+		Args:    cobra.ExactArgs(0),
+		Aliases: []string{"users"},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// Support `-R, --repo` override
+			opts.BaseRepo = f.BaseRepo
+			opts.HTTPClient = f.HttpClient
+
+			return runE(opts)
+		},
+	}
+
+	cmdutils.EnableRepoOverride(repoListCmd, f)
+
+	repoListCmd.Flags().StringVarP(&opts.OrderBy, "order", "o", "last_activity_at", "Return repositories ordered by id, created_at, or other fields")
+	repoListCmd.Flags().StringVarP(&opts.Sort, "sort", "s", "", "Return repositories sorted in asc or desc order")
+	repoListCmd.Flags().IntVarP(&opts.Page, "page", "p", 1, "Page number")
+	repoListCmd.Flags().IntVarP(&opts.PerPage, "per-page", "P", 30, "Number of items to list per page.")
+	return repoListCmd
+}
+
+func runE(opts *Options) error {
+	var err error
+	c := opts.IO.Color()
+
+	apiClient, err := opts.HTTPClient()
+	if err != nil {
+		return err
+	}
+
+	l := &gitlab.ListProjectsOptions{
+		OrderBy:        gitlab.String(opts.OrderBy),
+		MinAccessLevel: gitlab.AccessLevel(50), // Only projects you own
```

Comment:
I think it would be better to set this to Minimal access instead?

I tried to run this against my GitLab.com account. That showed me most of my projects(there is a [bug](https://gitlab.com/gitlab-org/gitlab/-/issues/219299)).

Then I tried to run against one of my self hosted instances, and it showed me nothing. When I go into Web UI I have 5 projects available, and I expected to see the same. The difference comes from the fact that I am just a developer on these projects.

And I feel that a lot of people might have such instances where they are only developers on a lot of projects and would want to see them, but can't.

Reply:
Yeah it tried it too on GitLab.com and it only show repos I own without including projects in my namespace

```
$ glab repo list

Showing 2 of 2 projects (Page 1 of 1)

glab-cli/test     
glab-cli/test-pv  
```

I suggest instead of restricting the result with the min access level, we should have flags to allow the user get the type of repos to return
```go
repoListCmd.Flags().BoolVarP(&opts.All, "all", "a", false, "list all projects on the instance")
repoListCmd.Flags().BoolVarP(&opts.Owned, "mine", "m", false, "limit by projects you own")
repoListCmd.Flags().BoolVar(&opts.Member, "member", false, "limit by projects which you are a member")
repoListCmd.Flags().BoolVar(&opts.Starred, "starred", false, "Get starred projects")
```